src/Makefile.am

```diff
@@ -493,6 +498,18 @@ libzcash_a_LDFLAGS = $(HARDENED_LDFLAGS)
 
 libzcash_a_CPPFLAGS += -DMONTGOMERY_OUTPUT
 
+# libsnark
+libsnark_a_SOURCES = \
+  libsnark/algebra/curves/alt_bn128/alt_bn128_g1.cpp \
+  libsnark/algebra/curves/alt_bn128/alt_bn128_g2.cpp \
+  libsnark/algebra/curves/alt_bn128/alt_bn128_init.cpp \
+  libsnark/algebra/curves/alt_bn128/alt_bn128_pairing.cpp \
+  libsnark/algebra/curves/alt_bn128/alt_bn128_pp.cpp \
+  libsnark/common/utils.cpp \
+  libsnark/common/profiling.cpp
+
+libsnark_a_CPPFLAGS = -fPIC -O2 -march=x86-64 -DMONTGOMERY_OUTPUT -DBINARY_OUTPUT -DNO_PT_COMPRESSION=1 -DMULTICORE -DCURVE_ALT_BN128 -fopenmp -DNO_PROCPS -static -DSTATIC -I./libsnark/
```

Comment:
libzcash for example doesn't have ``-march=x86-64``. Is that actually necessary, or should we just let gcc pick the arch by its normal rules?